test/noted/notes_test.exs

```diff
@@ -75,11 +79,56 @@ defmodule Noted.NotesTest do
       {:ok, tag} =
         attrs
         |> Enum.into(@valid_attrs)
-        |> Notes.create_tag()
+        |> Notes.create_tag!()
 
       tag
     end
 
+    def get_notes_tags(note, tag) do
+      Repo.one!(from n in NotesTags, where: n.note_id == ^note.id and n.tag_id == ^tag.id)
+    end
+
+    test "add_tag/3 add the tag to a note" do
+      telegram_id = System.unique_integer([:positive])
+      {:ok, user} = Accounts.create_user(%{telegram_id: telegram_id, telegram_data: %{}})
+
+      {:ok, note} = Notes.create_note(user.id, "test title", "test body", [])
+
+      tag_name = "test#{System.unique_integer([:positive])}"
+      {:ok, notes_tags} = Notes.add_tag(user.id, note.id, tag_name)
+
+      tag = Repo.get(Tag, notes_tags.tag_id)
+
+      assert notes_tags.tag_id == tag.id
+    end
+
+    test "add_tag/3 is idempotent" do
+      telegram_id = System.unique_integer([:positive])
+      {:ok, user} = Accounts.create_user(%{telegram_id: telegram_id, telegram_data: %{}})
+      {:ok, note} = Notes.create_note(user.id, "test title", "test body", [])
+      tag_name = "test#{System.unique_integer([:positive])}"
+
+      Notes.add_tag(user.id, note.id, tag_name)
+
+      assert Notes.add_tag(user.id, note.id, tag_name) == nil
```

Comment:
Change add_tag return value to always return :ok | {:error, changeset}
Never nil.

Then just `assert :ok = Notes.add_tag ..`